tasmota/xsns_78_as608.ino

```diff
@@ -0,0 +1,383 @@
+// Conditional compilation of driver
+#ifdef USE_AS608
+
+// Define driver ID
+#define XSNS_78  78
+
+#include <TasmotaSerial.h>
+#include <Adafruit_Fingerprint.h>
+
+
+bool as608selected = false;
+static uint8_t enrollstep = 0;
+static uint8_t ModellNumber = 0;
+
+Adafruit_Fingerprint *finger = NULL;
+TasmotaSerial *serial = NULL;
+
+uint8_t myFingerprint[sizeof(Adafruit_Fingerprint)];
+uint8_t myTasmotaSerial[sizeof(TasmotaSerial)];
+
+void as608init(){
+
+    if (PinUsed(GPIO_AS608_RX) && PinUsed(GPIO_AS608_TX)){
+
+        serial = new (myTasmotaSerial) TasmotaSerial(Pin(GPIO_AS608_RX), Pin(GPIO_AS608_TX), 0);
+        finger = new (myFingerprint) Adafruit_Fingerprint(serial, 0);
+
+        finger->begin(57600);
+
+        if (finger->verifyPassword()){
+          finger->getTemplateCount();
+          as608selected = true;
+          snprintf_P(log_data, sizeof(log_data), PSTR(D_LOG_LOG "AS60x found. %i fingerprints stored."), finger->templateCount);
+        }else{
+          snprintf_P(log_data, sizeof(log_data), PSTR(D_LOG_LOG "AS60x not found! Got data: %02X %02X %02X %02X %02X %02X %02X %02X %02X %02X %02X %02X"), finger->mydata[0], finger->mydata[1], finger->mydata[2], finger->mydata[3], finger->mydata[4], finger->mydata[5], finger->mydata[6], finger->mydata[7], finger->mydata[8], finger->mydata[9], finger->mydata[10], finger->mydata[11]);
+        }
+        AddLog(LOG_LEVEL_INFO);
+    }
+
+}
+
+uint8_t as608Enroll(uint8_t nr){
+     // set Variable to 1rst step
+    if (!enrollstep){
+        enrollstep = 1;
+        ModellNumber = nr;
+    }
+    return true;
+}
+
+int deleteFingerprint(uint8_t id) {
+  uint8_t p = -1;
+
+  p = finger->deleteModel(id);
+
+  if (p == FINGERPRINT_OK) {
+    snprintf_P(log_data, sizeof(log_data), PSTR(D_LOG_LOG "Delete ok"));
+    AddLog(LOG_LEVEL_INFO);
```

Comment:
Use `AddLog_P2(LOG_LEVEL_INFO, PSTR(D_LOG_LOG "Delete ok"));` instead. It makes only one call and shorter code.

Reply:
No Problem. I fix it next days.